src/functions.jl

```diff
@@ -625,16 +658,11 @@ for T in (:Taylor1, :TaylorN)
                 return nothing
             end
             zero!(c, k)
-            if $T == Taylor1
-                @inbounds c[k] = k * a[k] * c2[0]
-            else
-                @inbounds mul!(c[k], k * a[k], c2[0])
-            end
-            @inbounds for i = 1:k-1
+            @inbounds for i = 0:k-1
                 if $T == Taylor1
                     c[k] += (k-i) * a[k-i] * c2[i]
                 else
-                    mul!(c[k], (k-i) * a[k-i], c2[i])
+                    mul!(c[k], a[k-i], c2[i], scalar=k-i)
                 end
             end
```

Comment:
Same comment about order of `if` and `for`

Reply:
Done 👍